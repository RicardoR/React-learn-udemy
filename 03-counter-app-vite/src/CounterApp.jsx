 import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {

    const [counter, updateCounter] = useState(value);

    const updateValue = (val) => {
        updateCounter(() => counter + val )
    };

    return (
        <>
            <h1> CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={() => updateValue(1)}>+1</button>
            <button onClick={() => updateValue(2)}>+2</button>
            <button onClick={() => updateValue(-1)}>-1</button>
            <button onClick={() => updateValue(-2)}>-2</button>
        </>
    );
};


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};
