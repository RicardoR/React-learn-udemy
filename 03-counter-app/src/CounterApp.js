import PropTypes from 'prop-types';
import { Fragment, useState } from 'react';

const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);
  
  // By default setCaunter retrive the value from the state
  const decrease = () => setCounter((c) => c - 1); 
  const increase = () => setCounter(counter + 1);
  const reset = () => setCounter(value);

  return (
    <Fragment>
      <h1>CounterApp</h1>
      <button id='resetBtn' onClick={reset}>Reset</button>
      <div className="counter-controls">
        <button id='decreaseBtn' onClick={decrease}>- 1</button>
        <p> {counter} </p>
        <button id='increaseBtn' onClick={increase}>+ 1</button>
      </div>
    </Fragment>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
    value: 5,
}

export default CounterApp;