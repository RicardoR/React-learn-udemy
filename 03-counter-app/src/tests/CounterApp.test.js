import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('<CounterApp />', () => {
    let wrapper;
    let increaseBtn;
    let decreaseBtn;
    let resetBtn;

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
        expect(wrapper).toMatchSnapshot();
        increaseBtn = wrapper.find('#increaseBtn');
        decreaseBtn = wrapper.find('#decreaseBtn');
        resetBtn = wrapper.find('#resetBtn');
    });

    test('should display 5 as default value', () => {
        expect(wrapper.find('p').text().trim()).toBe('5');
    });

    test('should display the value passed to the component', () => {
        const ten = 10;
        const wrapperNew = shallow(<CounterApp value={ten} />);

        expect(wrapperNew).toMatchSnapshot();
        expect(wrapperNew.find('p').text().trim()).toBe(ten.toString());
    });

    test('should increase the value when increaseBtn is clicked', () => {
        increaseBtn.simulate('click');
        expect(wrapper.find('p').text().trim()).toBe('6');
    });
    
    test('should decrease the value when decreaseBtn is clicked', () => {
        decreaseBtn.simulate('click');
        expect(wrapper.find('p').text().trim()).toBe('4');
    });

    test('should reset de value when resetBtn is clicked', () => {
        increaseBtn.simulate('click');
        expect(wrapper.find('p').text().trim()).toBe('6');
        resetBtn.simulate('click');
        expect(wrapper.find('p').text().trim()).toBe('5');
    });
});