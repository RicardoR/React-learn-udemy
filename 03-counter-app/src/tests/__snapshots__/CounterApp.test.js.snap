// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp /> should decrease the value when decreaseBtn is clicked 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <button
    id="resetBtn"
    onClick={[Function]}
  >
    Reset
  </button>
  <div
    className="counter-controls"
  >
    <button
      id="decreaseBtn"
      onClick={[Function]}
    >
      - 1
    </button>
    <p>
       
      5
       
    </p>
    <button
      id="increaseBtn"
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
</Fragment>
`;

exports[`<CounterApp /> should display 5 as default value 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <button
    id="resetBtn"
    onClick={[Function]}
  >
    Reset
  </button>
  <div
    className="counter-controls"
  >
    <button
      id="decreaseBtn"
      onClick={[Function]}
    >
      - 1
    </button>
    <p>
       
      5
       
    </p>
    <button
      id="increaseBtn"
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
</Fragment>
`;

exports[`<CounterApp /> should display the value passed to the component 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <button
    id="resetBtn"
    onClick={[Function]}
  >
    Reset
  </button>
  <div
    className="counter-controls"
  >
    <button
      id="decreaseBtn"
      onClick={[Function]}
    >
      - 1
    </button>
    <p>
       
      5
       
    </p>
    <button
      id="increaseBtn"
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
</Fragment>
`;

exports[`<CounterApp /> should display the value passed to the component 2`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <button
    id="resetBtn"
    onClick={[Function]}
  >
    Reset
  </button>
  <div
    className="counter-controls"
  >
    <button
      id="decreaseBtn"
      onClick={[Function]}
    >
      - 1
    </button>
    <p>
       
      10
       
    </p>
    <button
      id="increaseBtn"
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
</Fragment>
`;

exports[`<CounterApp /> should increase the value when increaseBtn is clicked 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <button
    id="resetBtn"
    onClick={[Function]}
  >
    Reset
  </button>
  <div
    className="counter-controls"
  >
    <button
      id="decreaseBtn"
      onClick={[Function]}
    >
      - 1
    </button>
    <p>
       
      5
       
    </p>
    <button
      id="increaseBtn"
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
</Fragment>
`;

exports[`<CounterApp /> should reset de value when resetBtn is clicked 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <button
    id="resetBtn"
    onClick={[Function]}
  >
    Reset
  </button>
  <div
    className="counter-controls"
  >
    <button
      id="decreaseBtn"
      onClick={[Function]}
    >
      - 1
    </button>
    <p>
       
      5
       
    </p>
    <button
      id="increaseBtn"
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
</Fragment>
`;
